### PR TITLE
Beer Lovers Release

### PR DESCRIPTION
**Pull request by manq2010**
Beer Lovers Application
---

In this release, the following was done:
- [x] Added data from [Punk API](https://punkapi.com/documentation/v2)
- [x] Used [Behance](https://www.behance.net/sakwadesignstudio) for design guidelines
- [x] Created two pages, Main and Beer
- [x] Create units test and integrated tests for functions 
- [x] Implemented `scrollToTop`, `theToggle` and `navbar` components
- [x] Link to the project [presentation](https://drive.google.com/file/d/1pu-ZKbK1hU9rHcgt9qHP98jw5AWZlhrc/view?usp=sharing)

### CODE REVIEW

File: src/pages/main/MainPage.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        | Beer Lovers Application
+      </title>
+      <meta name="description" content="Beer lovers application" />
+    </Helmet>
+    <Navbar />
+    <Beers />
+    <Contact />
+    <Footer />
+
+  </div>
```
Comment (omar-muhamad) [inline src/pages/main/MainPage.jsx:24]: - [ ] s per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Layout%3A%20composition%20and%20space%20between%20elements.), For the home page, Please make the beers types shaped in a two-column grid and the background color is staggered like a chessboard to match the given design.

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

File: README.md [hunk 2]
```diff
@@ -184,6 +184,7 @@ Give a ⭐️ if you like this project!
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
 - Thank you [Microverse](https://www.microverse.org/) for the project
+- Also [Punk Api](https://punkapi.com/documentation/v2) for the API and documentation
```
Comment (omar-muhamad) [inline README.md]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.), At the `Acknowledgements` section, Please gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

File: src/__tests__/components.test.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          <Router location={history.location} navigator={history}>
+            <AppRoutes />
+          </Router>
+        </ThemeProvider>
+      </Provider>,
+    );
+
+    fireEvent.click(screen.getByTestId(/Beer Lovers/));
+    expect(history.location.pathname).toEqual('/');
+  });
```
Comment (omar-muhamad) [inline src/__tests__/components.test.js:47]: - [ ] You only added screenshot tests for your components, Please also add unit tests for some pure functions in Redux code and integration tests using React Testing Library mocking data fetching and Redux Store connection.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @manq2010,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] It would be better professionally if you rename the `features` directory to `components`.

image

- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__tests__/components.test.js (hunk 3 above)
Comment (manq2010) [reply src/__tests__/components.test.js:47]: https://github.com/manq2010/beer-lovers/blob/22948f770288591531dfc6c66110a675d722d6f0/src/__tests__/components.test.js#L1-L163

Please note the test file has all the required test

File: src/pages/main/MainPage.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        {headerData.name}
+        {' '}
+        | Beer Lovers Application
+      </title>
+      <meta name="description" content="Beer lovers application" />
+    </Helmet>
+    <Navbar />
+    <Beers />
+    <Contact />
+    <Footer />
```
Comment (fobadara) [inline src/pages/main/MainPage.jsx:22]: [REQUIRED] As suggested by the previous reviewer, kindly make the beer types shaped in a two-column grid and the background colour is staggered like a chessboard to match the given design. When you look at Cindy'sdesign you notice that there are no spaces. between the items.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @manq2010, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @manq2010 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

clapping-leonardo-dicaprio


## Highlights
All linters are passing :heavy_check_mark: 
Great effort in making the required changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sinansevgi — COMMENTED** [review]
Hello @manq2010 👋,

_This is an additional review as a result of the second opinion request._

- It appears that your project has been approved, and the issue has been resolved. If you require further clarification on any of the reviews, please reply to this message and tag me.

Keep rocking 🚀!

**Pull request merged**